templates/terraform/resource.erb

```diff
@@ -189,20 +189,26 @@ func resource<%= resource_name -%>Create(d *schema.ResourceData, meta interface{
     }
 <%    end -%>
 <%  end -%>
+    billingProject := ""
+
 <%  if has_project -%>
     project, err := getProject(d, config)
     if err != nil {
         return err
     }
+    billingProject = project
 <%  end -%>
+
 <%  if object.supports_indirect_user_project_override -%>
-    var project string
     if parts := regexp.MustCompile(`projects\/([^\/]+)\/`).FindStringSubmatch(url); parts != nil {
-        project = parts[1]
+        billingProject = parts[1]
     }
 <%  end -%>
+
+    billingProject, _ = getBillingProject(d, config)
```

Comment:
This will overwrite the direct/indirect project even if `""`+error is returned. I believe we want to check if the error is nil, eg:

```suggestion
    // err == nil indicates that the billing_project value was found
    if bp, err := getBillingProject(d, config); err == nil {
      billingProject = bp
    }
```

Reply:
fixed